uisdk/src/braintree/java/com/karhoo/uisdk/screen/booking/checkout/payment/braintree/BraintreePaymentActivity.kt

```diff
@@ -53,7 +54,10 @@ class BraintreePaymentActivity : BaseActivity(), DropInListener {
     override fun onDropInFailure(error: Exception) {
         finishActivity(if (isGuest()) BraintreePaymentView.REQ_CODE_BRAINTREE_GUEST else BraintreePaymentView.REQ_CODE_BRAINTREE)
         setResult(RESULT_OK, Intent().apply {
-            putExtra(BRAINTREE_ACTIVITY_DROP_IN_RESULT_ERROR, KarhooError.fromCustomError("", error.message!!, error.localizedMessage!!))
+            if(error !is UserCanceledException)
```

Comment:
Why not go with `if error is UserCanceled { putExtra(UserCanceledError) } else { putExtra(Error) }`? Same thing logically speaking, but the default is usually in the last `else`